pkg/flink/app/application_test.go

```diff
@@ -278,3 +279,86 @@ func (s *ApplicationTestSuite) TestPanicRecovery() {
 	cupaloy.SnapshotT(s.T(), actual)
 	require.Equal(s.T(), 1, callCount)
 }
+
+func (s *ApplicationTestSuite) TestPanicRecoveryWithLimitWhenLimitExceeded() {
+	// Given
+	recoverCount := 5
+	callCount := 0
+	s.app.reportUsage = func() {
+		callCount++
+	}
+	s.inputController.EXPECT().GetUserInput().Times(recoverCount).Do(func() {
+		panic("err in repl")
+	})
+	s.statementController.EXPECT().CleanupStatement().Times(recoverCount)
+
+	// When
+	actual := test.RunAndCaptureSTDOUT(s.T(), func() {
+		run := utils.NewPanicRecovererWithLimit(recoverCount, 3*time.Second)
+		for i := 0; i <= recoverCount; i++ {
```

Comment:
Did you mean to run this `recoverCount + 1` times?
```suggestion
		for i := 0; i < recoverCount; i++ {
```

Reply:
Yep I did. If I recover 5 times, I have to run it 6 so it panics on the last one